### ISSUE TITLE
🐛 `npx @biomejs/biome migrate` command crashes

### ISSUE THREAD

**Issue by sparrowsl** [biomejs/biome#3179]
### Environment information

```block
CLI:
  Version:                      1.8.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.14.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.7.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

`npx @biomejs/biome migrate` 

```bash
Biome encountered an unexpected error                                                                                                                                  
This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:
                                                                         
Source Location: /home/runner/work/biome/biome/crates/biome_rowan/src/ast/mod.rs:686:41 
Thread Name: main                                             
Message: Malformed list, node expected but found token COMMA@1621..1622 "," [] [] instead. You must add missing markers for missing elements.
```

### Expected result

It should update the biome.json file or something positive on what next to do, but not an error or bug report like this.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Is it possible that you have a leading comma in your configuration file, and that you enabled it in the configuration file?

**Comment by sparrowsl** [comment]
I have it in the `javascript.formatter` property section, and it is `"trailingCommas": "es5"`

**Comment by Sec-ant** [comment]
I can reproduce from your repo. I narrowed it down and found it was triggered by the promotion of the once nursery rule `useImportRestrictions` when it is not the last rule in the nursery group. In your example it is followed by `useImportRestrictions`. When I put it to the last rule of that group, or delete it from the group, the migration command suceeds.
